kplclient_test.go

```diff
@@ -25,3 +25,33 @@ func TestHappy(t *testing.T) {
 	//wait before shutting down
 	time.Sleep(20 * time.Second)
 }
+
+func TestNegative(t *testing.T) {
+	//Enabled it when testing for negative scenario. Pass invalid credential to AWS
+	t.SkipNow()
+	//give the test harness a chance to boot
+	time.Sleep(10 * time.Second)
+
+	c := NewKPLClient("127.0.0.1", "3000")
+
+	c.ErrPort = ":3001"
+	c.ErrHandler = func(data string) {
+		if data != "test" {
+			t.Error("Expected test", "Got", data)
+			t.Fail()
+		}
+	}
+
+	err := c.Start()
+	if err != nil {
+		t.Error(err)
+	}
+
+	err = c.PutRecord("test")
```

Comment:
Does this actually talk to Kinesis? 😬

Reply:
Yeah. It does. I promise. Testing this only took a lot of time.

Reply:
lol, I believe you. I'm grimacing because unit tests shouldn't typically talk to external services, that would make them integration tests... We should test here the expectations of kplclient (e.g. functions are called/not called, error handling, etc.), not the expectations of kplserver.

Reply:
Yes. It is more like integration test.